app/models/spree/flow_io_variant_decorator.rb

```diff
@@ -50,15 +50,16 @@ def remove_experience_from_product(exp_key, product)
     end
 
     # upload product variant to Flow's Product Catalog
-    def sync_product_to_flow
+    def sync_product_to_flow(force_sync = false)
```

Comment:
What does force sync represent?

Reply:
This **force_sync** param would be use to help us force the synchrornization of some products that does not match the minimum requirements by Flow (price more than 0 or having country of origin).

Right now, for example we have the **Ring Sizer**, which is not being charged at all; However, we still want to offer it on Flow countries. In order to do so, we need to force that product's synchronization into Flow and then set an Euro price for it using price books inside Flow.

I also realized that maybe I can rearrange this better, by separating this method into several ones and have a new one named after (force_sync_product_to_flow) that would skip validations.

Thanks for asking

Reply:
> I also realized that maybe I can rearrange this better, by separating this method into several ones and have a new one named after (force_sync_product_to_flow) that would skip validations.

Yep, I agree! It is a code smell to pass [boolean as a parameter](https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3) as it is a sign of a method doing too much. 
There is also a [check in reek](https://github.com/troessner/reek/blob/master/docs/Boolean-Parameter.md) but it is disabled now, maybe we should enable it?